src/simulators.jl

```diff
@@ -202,14 +217,27 @@ function simulate!(sys,
     neighbors = find_neighbors(sys, sys.neighbor_finder; n_threads=n_threads)
     run_loggers!(sys, neighbors, 0, run_loggers; n_threads=n_threads)
 
+    using_constraints = (length(sys.constraints) > 0)
+
+    ca_coord_storage = similar(sys.coords)
+
     for step_n in 1:n_steps
         accels_t = accelerations(sys, neighbors; n_threads=n_threads)
 
         sys.velocities += accels_t .* sim.dt
+        
+        if length(sys.constraints) > 0
+            ca_coord_storage .= sys.coords
+        end
 
-        old_coords = copy(sys.coords)
         sys.coords += sys.velocities .* sim.dt
-        apply_constraints!(sys, old_coords, sim.dt)
+
+        sys = apply_position_constraints!(sys, ca_coord_storage, n_threads=n_threads)
```

Comment:
In general you don't need to set `sys = ` here, even though `apply_position_constraints!` returns `sys`.